quil-py/src/instruction/mod.rs

```diff
@@ -0,0 +1,77 @@
+use quil_rs::instruction::Instruction;
+use rigetti_pyo3::{create_init_submodule, impl_repr, impl_str, py_wrap_union_enum};
+
+pub use self::{
+    arithmetic::{
+        PyArithmetic, PyArithmeticOperand, PyArithmeticOperator, PyBinaryLogic, PyBinaryOperand,
+        PyBinaryOperands, PyBinaryOperator,
+    },
+    calibration::{PyCalibration, PyMeasureCalibrationDefinition},
+    declaration::{PyDeclaration, PyScalarType, PyVector},
+    expression::{
+        PyExpression, PyExpressionFunction, PyFunctionCallExpression, PyInfixExpression,
+        PyInfixOperator,
+    },
+    frame::{PyAttributeValue, PyFrameAttributes, PyFrameDefinition, PyFrameIdentifier},
+    gate::{PyGate, PyGateDefinition, PyGateError, PyGateModifier, PyGateSpecification},
+    memory_region::{PyMemoryReference, PyMemoryRegion},
+    qubit::PyQubit,
+    waveform::{PyWaveform, PyWaveformDefinition},
+};
```

Comment:
I found it difficult to work in one large `instructions.rs` file, so I started breaking them down into groups. Categorizing can sometimes be arbitrary so these groupings are transparent to users of the crate.